PowerSwitch/Public/Exos/Get-ExosVlanConfig.ps1

```diff
@@ -50,49 +50,55 @@ function Get-ExosVlanConfig {
             $KeepGoing = $true
             continue
         }
 
         if ($KeepGoing) {
-            $EvalParams.Regex = [regex] '^enable\ eaps$'
+            # create vlan "(vlan-name)"
+            $EvalParams.Regex = [regex] "^create\ vlan\ `"(?<vlanname>.+?)`""
             $Eval = Get-RegexMatch @EvalParams
             if ($Eval) {
-                Write-Verbose "$VerbosePrefix $i`: eaps: enabled"
-                $ReturnObject.Enabled = $true
+                $VlanName = $Eval.Groups['vlanname'].Value
+                Write-Verbose "$VerbosePrefix $i`: vlan: name '$VlanName'"
+                $New = [Vlan]::new($VlanName)
+                $ReturnArray += $New
                 continue
             }
 
-
-            # create vlan "(vlan-name)"
-            $EvalParams.Regex = [regex] "^configure\ eaps\ (?<domain>.+?)\ mode\ (?<mode>.+)"
+            # configure vlan tag
+            $EvalParams.Regex = [regex] "^configure\ vlan\ `"?(?<vlanname>.+?)`"?\ tag\ (?<vlantag>.+)"
             $Eval = Get-RegexMatch @EvalParams
             if ($Eval) {
-                $New = [Vlan]::new($NAMEVARIABLE)
-
-
-                $Domain = $Eval.Groups['domain'].Value
-                $Mode = $Eval.Groups['mode'].Value
-                Write-Verbose "$VerbosePrefix $i`: eaps: domain '$Domain' mode: $Mode"
-                $DomainLookup = $ReturnObject.Domain | Where-Object { $_.Name -eq $Domain }
-                $DomainLookup.Mode = $Mode
+                $VlanName = $Eval.Groups['vlanname'].Value
+                $VlanTag = $Eval.Groups['vlantag'].Value
+                Write-Verbose "$VerbosePrefix $i`: vlan: name '$VlanName' tag: $VlanTag"
+                $VlanLookup = $ReturnArray | Where-Object { $_.Name -eq $VlanName }
+                $VlanLookup.Id = $VlanTag
                 continue
             }
 
-
-            # configure eaps <domain> mode <mode>
-            $EvalParams.Regex = [regex] "^configure\ eaps\ (?<domain>.+?)\ mode\ (?<mode>.+)"
+            # adding ports to vlan
+            $EvalParams.Regex = [regex] "^configure\ vlan\ `"?(?<vlanname>.+?)`"?\ add\ ports\ (?<portnumber>.+)\ (?<type>tagged|untagged)"
             $Eval = Get-RegexMatch @EvalParams
             if ($Eval) {
-                $Domain = $Eval.Groups['domain'].Value
-                $Mode = $Eval.Groups['mode'].Value
-                Write-Verbose "$VerbosePrefix $i`: eaps: domain '$Domain' mode: $Mode"
-                $DomainLookup = $ReturnObject.Domain | Where-Object { $_.Name -eq $Domain }
-                $DomainLookup.Mode = $Mode
+                $VlanName = $Eval.Groups['vlanname'].Value
+                $PortNumber = $Eval.Groups['portnumber'].Value
+                $Type = $Eval.Groups['type'].Value
+                Write-Verbose "$VerbosePrefix $i`: vlan: name '$VlanName' port: $PortNumber type: $Type"
+                $VlanLookup = $ReturnArray | Where-Object { $_.Name -eq $VlanName }
+                switch ($Type) {
```

Comment:
this is just adding the string of ports defined, we need to add an entry for each individual port. For Instance, instead of adding
`1:1-2, 1:4`
we need to add
`1:1` and `1:2` and `1:4`

I have a helper function in `/PowerSwitch/Private` call `Resolve-PortString`. It currently support Eos only, but we should add exos to it, you can take a look at it and see how it works, it's a bit tricky. I can help later this afternoon if needed, but I'm in meeting until 4. Once exos support is added, you would use it like this:
`$VlanLookup.TaggedPorts += Resolve-PortString -PortString $ThesePorts -SwitchType 'Exos'`